x-pack/filebeat/processors/decode_cef/decode_cef_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package decode_cef
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type testCase struct {
+	config  func() config
+	message string
+	fields  common.MapStr
+}
+
+var testCases = map[string]testCase{
+	"default": {
+		message: "CEF:0|Trend Micro|Deep Security Manager|1.2.3|600|User Signed In|3|src=10.52.116.160 suser=admin target=admin msg=User signed in from 2001:db8::5",
+		fields: common.MapStr{
+			"cef.version":                   0,
+			"cef.device.event_class_id":     "600",
+			"cef.device.product":            "Deep Security Manager",
```

Comment:
I think CEF's concept of "device" maps well with ECS' `observer`. Actually I think CEF is specifically the reason `observer` was initially called `device` in ECS.

However we don't have a field for product name in `observer`.

So I'd suggest we map

- `cef.device.vendor` to `observer.vendor`
- `cef.device.version` to `observer.version`

Then we could add a field to ECS for the product name, and map this one accordingly as well, WDYT?